### PR TITLE
[SAJC] Add title, screen element and radiobutton group to parameters

### PR DESCRIPTION
**Pull request by heichmann**
A new type of class-based application jobs is introduced. Here, the title of the parameter, the screen element which is used for the parameter and the radio button group are not set in the class any more, but in the application job catalog entry. Therefore, the application job catalog entry has to be extended.

### CODE REVIEW

**Comment by heichmann** [comment]
About the attribute 'delete' which was removed from structure ty_exit_classes: This attribute was not shipped to the customers yet, so it is no incompatible change. We decided that the delete exit class shall not be supported yet.

**Comment by wurzka** [comment]
I agree with @heichmann. Since `delete` was not shipped to the customers yet, we can perform this (incompatible) change.

File: file-formats/sajc/type/zif_aff_sajc_v1.intf.abap [hunk 1]
```diff
@@ -64,9 +61,38 @@ INTERFACE zif_aff_sajc_v1
...
+    BEGIN OF co_screen_element,
+      "! <p class="shorttext">None</p>
+      "! None
+      none        TYPE ty_screen_element VALUE ' ',
+      "! <p class="shorttext">Checkbox</p>
+      "! Checkbox
+      checkbox    TYPE ty_screen_element VALUE 'C',
+      "! <p class="shorttext">Radio Button</p>
+      "! Radio button
+      radiobutton TYPE ty_screen_element VALUE 'R',
```
Comment (schneidermic0) [inline file-formats/sajc/type/zif_aff_sajc_v1.intf.abap:91]: Minor: Does it make sense to change it to `radio_button`?

```suggestion
      radio_button TYPE ty_screen_element VALUE 'R',
```

File: file-formats/sajc/type/zif_aff_sajc_v1.intf.abap [hunk 2]
```diff
@@ -64,9 +61,38 @@ INTERFACE zif_aff_sajc_v1
...
+      "! <p class="shorttext">List Box</p>
+      "! List box
+      listbox     TYPE ty_screen_element VALUE 'L',
```
Comment (schneidermic0) [inline file-formats/sajc/type/zif_aff_sajc_v1.intf.abap:94]: Minor: Same here? `list_box`?

```suggestion
      list_box     TYPE ty_screen_element VALUE 'L',
```

File: file-formats/sajc/type/zif_aff_sajc_v1.intf.abap [hunk 3]
```diff
@@ -94,6 +123,12 @@ INTERFACE zif_aff_sajc_v1
       "! <p class="shorttext">Enabled By Parameter</p>
       "! Name of the boolean parameter which enables / disables the current parameter
       enabled_by_parameter TYPE ty_parameter_name,
+      "! <p class="shorttext">Screen Element</p>
+      "! Display of the parameter as screen element (radio button, checkbox, list box)
+      screen_element       TYPE ty_screen_element,
+      "! <p class="shorttext">Radio Button Group</p>
+      "! Name of the radio button group if the parameter is a radio button
+      radiobutton_group    TYPE ty_radiobutton_group,
```
Comment (schneidermic0) [inline file-formats/sajc/type/zif_aff_sajc_v1.intf.abap:131]: In dependency to the comments:

```suggestion
      radio_button_group    TYPE ty_radio_button_group,
```

File: file-formats/sajc/type/zif_aff_sajc_v1.intf.abap [hunk 4]
```diff
@@ -64,9 +61,38 @@ INTERFACE zif_aff_sajc_v1
...
   TYPES ty_section_name TYPE c LENGTH 10.
+  "! <p class="shorttext">Parameter Title Text</p>
+  "! Text of the parameter title
+  TYPES ty_parameter_title_text TYPE c LENGTH 255.
   "! <p class="shorttext">Title Text</p>
   "! Text of the title
   TYPES ty_title_text TYPE c LENGTH 120.
+  "! <p class="shorttext">Radio Button Group</p>
+  "! Name of the radio button group
+  TYPES ty_radiobutton_group TYPE c LENGTH 4.
```
Comment (schneidermic0) [inline file-formats/sajc/type/zif_aff_sajc_v1.intf.abap:72]:
```suggestion
  TYPES ty_radio_button_group TYPE c LENGTH 4.
```

File: file-formats/sajc/README.md [hunk 5]
```diff
@@ -1,5 +1,5 @@
 # SAJC File Format
 
 File | Cardinality | Definition | Schema | Example
-:--- | :---  | :--- | :--- | :---
-`<name>.sajc.json` | 1 | [`zif_aff_sajc_v1.intf.abap`](./type/zif_aff_sajc_v1.intf.abap) | [`sajc-v1.json`](./sajc-v1.json) | [`z_aff_example_sajc.sajc.json`](./examples/z_aff_example_sajc.sajc.json)
+:--- | :--- | :--- | :--- | :---
+`<name>.sajc.json` | 1 |  [`zif_aff_sajc_v1.intf.abap`](./type/zif_aff_sajc_v1.intf.abap)  | [`sajc-v1.json`](./sajc-v1.json) | [z_aff_example_sajc.sajc.json](./examples/z_aff_example_sajc.sajc.json)
```
Comment (schneidermic0) [inline file-formats/sajc/README.md:5]:
```suggestion
`<name>.sajc.json` | 1 | [`zif_aff_sajc_v1.intf.abap`](./type/zif_aff_sajc_v1.intf.abap) | [`sajc-v1.json`](./sajc-v1.json) | [`z_aff_example_sajc.sajc.json`](./examples/z_aff_example_sajc.sajc.json)
```

File: file-formats/sajc/type/zif_aff_sajc_v1.intf.abap (hunk 1 above)
Comment (heichmann) [reply file-formats/sajc/type/zif_aff_sajc_v1.intf.abap:91]: Yes, these changes make sense. I will change it.

File: file-formats/sajc/README.md (hunk 5 above)
Comment (schneidermic0) [reply file-formats/sajc/README.md:5]: I suggest to change it in this PR manually. 

The generator report must be fixed. See SAP/abap-file-formats-tools#300

File: file-formats/sajc/type/zif_aff_sajc_v1.intf.abap (hunk 1 above)
Comment (heichmann) [reply file-formats/sajc/type/zif_aff_sajc_v1.intf.abap]: I changed it.

File: file-formats/sajc/type/zif_aff_sajc_v1.intf.abap (hunk 2 above)
Comment (heichmann) [reply file-formats/sajc/type/zif_aff_sajc_v1.intf.abap]: I changed it.

File: file-formats/sajc/type/zif_aff_sajc_v1.intf.abap (hunk 3 above)
Comment (heichmann) [reply file-formats/sajc/type/zif_aff_sajc_v1.intf.abap]: I changed it.

File: file-formats/sajc/type/zif_aff_sajc_v1.intf.abap (hunk 4 above)
Comment (heichmann) [reply file-formats/sajc/type/zif_aff_sajc_v1.intf.abap]: I changed it.

File: file-formats/sajc/README.md (hunk 5 above)
Comment (heichmann) [reply file-formats/sajc/README.md]: I changed it.

**Review by schneidermic0 — APPROVED** [review]
Thanks, @heichmann. Looks good to me

**Review by albertmink — APPROVED** [review]
LGTM, UX review done.

**Pull request merged**